src/index.ts

```diff
@@ -1 +1,84 @@
-export const add = (a: number, b: number) => a + b;
+import {
+  getGeneratorForPrimes,
+  getPrime,
+  getRandomBigIntFromInterval,
+} from './utils.js';
+import { primalityTest } from 'miller-rabin-primality';
+
+export * from './utils.js';
+
+export class PublicKey {
+  primeP: bigint;
+  primeQ: bigint;
+  generator: bigint;
+  publicKey: bigint;
+
+  constructor(
+    primeP: bigint,
+    primeQ: bigint,
+    generator: bigint,
+    publicKey: bigint,
+  ) {
+    this.primeP = primeP;
+    this.primeQ = primeQ;
+    this.generator = generator;
+    this.publicKey = publicKey;
+  }
+}
+
+export class PrivateKey extends PublicKey {
+  privateKey: bigint;
+
+  constructor(
+    primeP: bigint,
+    primeQ: bigint,
+    generator: bigint,
+    publicKey: bigint,
+    privateKey: bigint,
+  ) {
+    super(primeP, primeQ, generator, publicKey);
+    this.privateKey = privateKey;
+  }
+}
+
+export class KeyPair {
+  publicKey: PublicKey;
+  privateKey: PrivateKey;
+
+  constructor(primeP: bigint, primeQ: bigint) {
+    if ((primeP - 1n) % primeQ !== 0n) {
+      throw new Error('Invalid: (p - 1) is not divisible by q');
+    }
+
+    const generator = getGeneratorForPrimes(primeP, primeQ);
+    const privateKey = getRandomBigIntFromInterval(1n, primeP);
+    const publicKey = (generator ^ privateKey) % primeP;
```

Comment:
```suggestion
    const publicKey = (generator ** privateKey) % primeP;
```
see: https://www.w3schools.com/js/js_operators.asp